File: src/dataset/c/collatz-conjecture_exercism-yorange1/gemini/GeminiFlash_collatz_conjecture_v3.c
#include "collatz_conjecture.h"

#define is_even(n) ((n & 1) == 0)

int steps(int start) {
  if (start <= 0) {
    return ERROR_VALUE;
  }

  int count = 0;
  unsigned int n = (unsigned int)start;

  while (n != 1) {
    if (is_even(n)) {
      n >>= 1;
    } else {
      if (n > (UINT_MAX - 1) / 3) {
        return ERROR_VALUE;
      }
      n = n * 3 + 1;
      if (n < start) {
          return ERROR_VALUE;
      }
    }
    count++;
  }

  return count;
}

File: src/dataset/typescript/collatz-conjecture_exercism-wackerow/gemini/GeminiFlash_collatz-conjecture_v3.ts
class CollatzConjecture {
    static steps(start: number): number {
        if (start <= 0 || !Number.isInteger(start)) {
            throw 'Only positive integers are allowed';
        }

        let count = 0;
        let n = start;

        while (n !== 1) {
            n = (n & 1) === 0 ? n >>> 1 : 3 * n + 1;
            count++;
        }

        return count;
    }
}

export default CollatzConjecture;